cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -80,13 +71,7 @@ public void running() {
       @Override
       public void failed(Service.State from, Throwable failure) {
         LOG.error("Program terminated with exception", failure);
-        if (propagateServiceError()) {
-          error(failure);
-        } else if (getState() != State.STOPPING) {
-          complete();
-        } else {
-          stop();
-        }
+        error(failure);
```

Comment:
Its `enum` in the `ProgramController` interface. The `getState()` method returns the state of the current ProgramController. We compare it with the state `State.STOPPING`. Since we are already in the `ProgramControllerServiceAdapter` which implements in the `ProgramController` interface, we don't need to qualify it as `ProgramController.State.STOPPING`